test/mix/phoenix_test.exs

```diff
@@ -3,6 +3,8 @@ defmodule Mix.PhoenixTest do
 
   test "base/0 returns the module base based on the Mix application" do
     assert Mix.Phoenix.base == "Phoenix"
+    Application.put_env(:phoenix, :phoenix_namespace, Phoenix.Sample.App)
```

Comment:
We need to revert this though just on the chance it can affect other tests. :) You can just use delete_env.